modules/inject/src/main/scala/io/janstenpickle/trace4cats/inject/EntryPoint.scala

```diff
@@ -7,27 +7,33 @@
 package io.janstenpickle.trace4cats.inject
 
 import cats.Applicative
+import cats.data.Kleisli
 import cats.effect.{Clock, Resource, Sync}
-import io.janstenpickle.trace4cats.{Span, ToHeaders}
 import io.janstenpickle.trace4cats.kernel.{SpanCompleter, SpanSampler}
 import io.janstenpickle.trace4cats.model.{SpanKind, TraceHeaders}
+import io.janstenpickle.trace4cats.{Span, ToHeaders}
 
 /** An entry point, for creating root spans or continuing traces that were started on another
   * system.
   */
 trait EntryPoint[F[_]] {
 
   /** Resource that creates a new root span in a new trace. */
-  def root(name: String): Resource[F, Span[F]] = root(name, SpanKind.Internal)
-  def root(name: String, kind: SpanKind): Resource[F, Span[F]]
+  def root(name: SpanName): Resource[F, Span[F]] = root(name, SpanKind.Internal)
+  def root(name: SpanName, kind: SpanKind): Resource[F, Span[F]]
 
   /** Resource that attempts to creates a new child span but falls back to a new root
     * span as with `root` if the headers do not contain the required values. In other words, we
     * continue the existing span if we can, otherwise we start a new one.
     */
-  def continueOrElseRoot(name: String, headers: TraceHeaders): Resource[F, Span[F]] =
+  def continueOrElseRoot(name: SpanName, headers: TraceHeaders): Resource[F, Span[F]] =
     continueOrElseRoot(name, SpanKind.Server, headers)
-  def continueOrElseRoot(name: String, kind: SpanKind, headers: TraceHeaders): Resource[F, Span[F]]
+  def continueOrElseRoot(name: SpanName, kind: SpanKind, headers: TraceHeaders): Resource[F, Span[F]]
+
+  def toReader: ResourceKleisli[F, SpanParams, Span[F]] =
```

Comment:
This name is also obsolete.

Reply:
The rest is fine. We can merge it.

Reply:
Ah nice catch. Thanks for all the feedback!